### PR TITLE
Add cancel-in-progress to workflows

### PR DESCRIPTION
**Pull request by skjnldsv**
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency

### CODE REVIEW

File: workflow-templates/lint-php.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@ on:
 permissions:
   contents: read
 
+concurrency: 
+  group: lint-php-${{ github.head_ref || github.run_id }}
```
Comment (nickvergessen) [inline workflow-templates/lint-php.yml:20]:
```suggestion
  group: lint-php-${{ github.head_ref || github.run_id }}-${{ matrix.php-versions }}
```

File: workflow-templates/phpunit-mysql.yml [hunk 2]
```diff
@@ -16,6 +16,10 @@ on:
 permissions:
   contents: read
 
+concurrency: 
+  group: phpunit-mysql-${{ github.head_ref || github.run_id }}
```
Comment (nickvergessen) [inline workflow-templates/phpunit-mysql.yml:20]:
```suggestion
  group: phpunit-mysql-${{ github.head_ref || github.run_id }}-${{ matrix.server-versions }}-${{ matrix.php-versions }}
```

File: workflow-templates/phpunit-oci.yml [hunk 3]
```diff
@@ -16,6 +16,10 @@ on:
 permissions:
   contents: read
 
+concurrency: 
+  group: phpunit-oci-${{ github.head_ref || github.run_id }}
```
Comment (nickvergessen) [inline workflow-templates/phpunit-oci.yml:20]:
```suggestion
  group: phpunit-oci-${{ github.head_ref || github.run_id }}-${{ matrix.server-versions }}-${{ matrix.php-versions }}
```

File: workflow-templates/phpunit-pgsql.yml [hunk 4]
```diff
@@ -16,6 +16,10 @@ on:
 permissions:
   contents: read
 
+concurrency: 
+  group: phpunit-pgsql-${{ github.head_ref || github.run_id }}
```
Comment (nickvergessen) [inline workflow-templates/phpunit-pgsql.yml:20]:
```suggestion
  group: phpunit-pgsql-${{ github.head_ref || github.run_id }}-${{ matrix.server-versions }}-${{ matrix.php-versions }}
```

File: workflow-templates/phpunit-sqlite.yml [hunk 5]
```diff
@@ -16,6 +16,10 @@ on:
 permissions:
   contents: read
 
+concurrency: 
+  group: phpunit-sqlite-${{ github.head_ref || github.run_id }}
```
Comment (nickvergessen) [inline workflow-templates/phpunit-sqlite.yml:20]:
```suggestion
  group: phpunit-sqlite-${{ github.head_ref || github.run_id }}-${{ matrix.server-versions }}-${{ matrix.php-versions }}
```

**Comment by ChristophWurst** [comment]
So, I guess in words this means that if you push twice in a short time period and the previous job hasn't finished yet, the new job will kill/replace it?

**Comment by skjnldsv** [comment]
> So, I guess in words this means that if you push twice in a short time period and the previous job hasn't finished yet, the new job will kill/replace it?

exactly

**Comment by nickvergessen** [comment]
Can you do the psalm tempalte too?

**Comment by skjnldsv** [comment]
LGTM :ship:

**Review by nickvergessen — APPROVED** [review]

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/nextcloud-browser-storage/pull/195 + https://github.com/nextcloud/nextcloud-browser-storage/actions/runs/3256220112/jobs/5346388533

> This job was skipped

Bub of this new configuration?

**Pull request merged**